src/cogent3/core/new_moltype.py

```diff
@@ -336,6 +342,7 @@ class MolType:
     _degen: new_alphabet.CharAlphabet = dataclasses.field(init=False)
     _degen_gapped: new_alphabet.CharAlphabet = dataclasses.field(init=False)
     _colors: dict[str, str] = dataclasses.field(init=False)
+    _strictly_upper_case: bool = dataclasses.field(init=False)
```

Comment:
add refactor comment to add arg like `valid=True` for when users know that is no need to check validity